runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -90,13 +91,13 @@
    *
    * @param <T> the type of elements that can be added to this bundle
    */
-  public static interface UncommittedBundle<T> {
+  interface UncommittedBundle<T> {
     /**
      * Returns the PCollection that the elements of this {@link UncommittedBundle} belong to.
      */
+    @Nullable
```

Comment:
It would be nice if these bundles weren't associated with the pcollection cause having nulls here will need to have people understand the special case of the bundle being the input to a root transform. Ditto on CommittedBundle